Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -1544,14 +1544,29 @@ namespace OpenInfraPlatform {
 						carve::geom3d::Vector& first = loop.front();
 						carve::geom3d::Vector& last = loop.back();
 
+						//bool areEqual = GeometrySettings::areEqual(loop.front(), loop.back());
+						if (GeometrySettings::areEqual(loop.front(), loop.back()))
+						{
+							loop.pop_back();
+							continue;
+						}
+						
+						/*
+						if (GeometrySettings::areEqual(first, last) == true)
+						{
+							loop.pop_back();
+							continue;
+						}*/
+
+						/*
 						if (abs(first.x - last.x) < 0.00000001) {
 							if (abs(first.y - last.y) < 0.00000001) {
 								if (abs(first.z - last.z) < 0.00000001) {
 									loop.pop_back();
 									continue;
 								}
 							}
-						}
+						}*/

```

Comment:
If it's working, no need for commented out code.

Reply:
Agree. Commented code generally only results in confusion and tends to be forgotten at some point.

Reply:
@pjanck , @hechth It's not working yet. I asked @Elvira2227 to push code unfinished, so I could find a mistake in the code. Thus we left code commented until we haven't found solution to our problem, so we could get back to previous version of the code.

Reply:
Right, I understand. If you need help, ping me.

Reply:
Resolved: 779b8e1